pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -285,18 +261,19 @@ def __transfer_distrib_spawn_state_on_fit_end(self, trainer: "pl.Trainer", resul
             last_path = re.sub(".ckpt", ".tmp_end.ckpt", best_model_path)
             self.checkpoint_io.save_checkpoint(state_dict, last_path)
 
-        # todo, pass complete checkpoint as state dictionary
-        self.mp_queue.put(best_model_path)
-        self.mp_queue.put(last_path)
-        self.mp_queue.put(results)
         # adds the `callback_metrics` to the queue
-        # TODO: Remove the if in v1.7
+        extra = _ExtraQueue()
         if is_overridden("add_to_queue", self.lightning_module):
-            self.lightning_module.add_to_queue(self.mp_queue)
+            # TODO: Remove the if in v1.7
+            self.lightning_module.add_to_queue(extra)
         else:
-            self.add_to_queue(trainer, self.mp_queue)
+            self.add_to_queue(trainer, extra)
+
+        return best_model_path, last_path, results, extra
```

Comment:
If this is not going to be removed soon in a follow-up, I'd suggest a NamedTuple here, it'll aid with the typing too

Reply:
Agreed, NamedTuple/TypedDict/Dataclass will make this more explicit

Reply:
Ah true, I was thinking about this too. 
Any of the three options containers would be ok I guess, as long as it's pickleable. 
What's the preference?

Reply:
+1 namedTuple

Reply:
going with namedTuple for now